src/snowflake/snowpark/dataframe.py

```diff
@@ -165,21 +170,44 @@ def _get_unaliased(col_name: str) -> List[str]:
     return unaliased
 
 
-def _alias_if_needed(df: "DataFrame", c: str, prefix: str, common_col_names: List[str]):
+def _alias_if_needed(
+    df: "DataFrame",
+    c: str,
+    prefix: Optional[str],
+    suffix: Optional[str],
+    common_col_names: List[str],
+):
     col = df.col(c)
-    unquoted = c.strip('"')
+    unquoted_col_name = c.strip('"')
     if c in common_col_names:
-        return col.alias(f'"{prefix}{unquoted}"')
+        if suffix:
+            column_case_insensitive = is_snowflake_quoted_id_case_insensitive(c)
+            suffix_unqouted_case_insensitive = (
+                is_snowflake_unquoted_suffix_case_insensitive(suffix)
+            )
+            return col.alias(
+                f'"{unquoted_col_name}{suffix.upper()}"'
+                if column_case_insensitive and suffix_unqouted_case_insensitive
+                else f'''"{unquoted_col_name}{escape_quotes(suffix.strip('"'))}"'''
+            )
+        return col.alias(f'"{prefix}{unquoted_col_name}"')
     else:
-        return col.alias(f'"{unquoted}"')
+        return col.alias(f'"{unquoted_col_name}"')
 
 
 def _disambiguate(
     lhs: "DataFrame",
     rhs: "DataFrame",
     join_type: JoinType,
     using_columns: List[str],
+    *,
+    lsuffix: str = "",
+    rsuffix: str = "",
 ) -> Tuple["DataFrame", "DataFrame"]:
+    if lsuffix == rsuffix and lsuffix:
+        raise ValueError(
+            f"'lsuffix' and 'rsuffix' must be different if they're not empty. You set {lsuffix!r} to both."
```

Comment:
is the single quote in "they're" going to cause problems? If so, you can change it to "they are".

Reply:
No the single quote doesn't cause a problem.